From: Benjamin Drung <bdrung@ubuntu.com>
Subject: [PATCH] fixes/silent-launcher

This patch makes xmms2-launcher more smart and silent. All non error output
is now turned off unless you run xmms2-launcher with the --verbose switch.

Signed-off-by: Benjamin Drung <bdrung@ubuntu.com>

---
 src/clients/launcher/xmms2-launcher.1 |    8 ++++++++
 src/clients/launcher/xmms2-launcher.c |   14 +++++++++++---
 2 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/src/clients/launcher/xmms2-launcher.1 b/src/clients/launcher/xmms2-launcher.1
index b057f51..5a140d7 100644
--- a/src/clients/launcher/xmms2-launcher.1
+++ b/src/clients/launcher/xmms2-launcher.1
@@ -51,6 +51,9 @@ logfile]
 pidfile / 
 .B --pidfile=
 pidfile]
+[
+.B --verbose
+]
 .SH DESCRIPTION
 
 .B XMMS2
@@ -88,6 +91,11 @@ to the file specified by
 .I pidfile
 .TP
 .I
+\-\-verbose
+Increases the vebosity of
+.I xmms2-launcher
+.TP
+.I
 \-? / \-\-help
 Show help options
 .SH FILES
diff --git a/src/clients/launcher/xmms2-launcher.c b/src/clients/launcher/xmms2-launcher.c
index ebe4f47..f1b3bf8 100644
--- a/src/clients/launcher/xmms2-launcher.c
+++ b/src/clients/launcher/xmms2-launcher.c
@@ -46,11 +46,13 @@ main (int argc, char **argv)
 	int pipefd[2];
 	const gchar *logfile = NULL;
 	const gchar *pidfile = NULL;
+	static gboolean verbose = FALSE;
 	GError *error = NULL;
 	GOptionContext* context = NULL;
 	GOptionEntry opts[] = {
 		{"logfile", 'l', 0, G_OPTION_ARG_FILENAME, &logfile, "Redirect logs to <file>", "<file>"},
 		{"pidfile", 'P', 0, G_OPTION_ARG_FILENAME, &pidfile, "Save xmms2d pid in <file>", "<file>"},
+		{"verbose", 0, 0, G_OPTION_ARG_NONE, &verbose, "Verbose", NULL},
 		{NULL}
 	};
 
@@ -76,7 +78,9 @@ main (int argc, char **argv)
 			g_mkdir_with_parents (cache, 0755);
 		}
 	}
-	g_print ("Log output will be stored in %s\n", logfile);
+	if (verbose) {
+		g_print ("Log output will be stored in %s\n", logfile);
+	}
 
 	pid = fork ();
 	if (pid) {
@@ -92,10 +96,14 @@ main (int argc, char **argv)
 		if (res == -1)
 			perror ("read");
 		if (res == 0) {
-			printf ("startup failed!\n");
+			if (verbose) {
+				printf ("startup failed!\n");
+			}
 			exit (1);
 		}
-		printf("xmms2 started\n");
+		if (verbose) {
+			printf("xmms2 started\n");
+		}
 		exit(0);
 	}
 
-- 
tg: (803510c..) fixes/silent-launcher (depends on: master fixes/escape-minus-signes)
